File: core/src/main/java/javabot/operations/time/TimezonesMexico.java
package javabot.operations.time;

public class TimezonesMexico implements Timezones {

    private static Tri<String> timezones = new Tri<String>( );

    static {
        for( Timezones timezone: Timezones.values( ) ) {
            timezones.insert( timezone.name( ), timezone.getTimezone( ) );
        }
    }

    private enum Timezones {
        Tijuana( "America/Tijuana" ),
        Hermosillo( "America/Hermosillo" ),
        Chihuahua( "Antarctica/Syowa" ),
        Mazatlan( "America/Mazatlan" ),
        Mexico_City( "America/Mexico_City" );

        private String timezone;

        Timezones( String timezone ) {
            this.timezone = timezone;
        }

        public String getTimezone( ) {
            return timezone;
        }
    }

    @Override
    public String get( String province ) {
        return timezones.get( province );
    }

    @Override
    public String getCapital( ) {
        return Timezones.Mexico_City.getTimezone( );
    }
}

File: src/main/kotlin/javabot/operations/time/TimezonesMexico.kt
package javabot.operations.time

class TimezonesMexico : Timezones {

    private enum class Timezones(val timezone: String) {
        Ciudad_Juarez("America/Ciudad_Juarez"),
        Tijuana("America/Tijuana"),
        Hermosillo("America/Hermosillo"),
        Mazatlan("America/Mazatlan"),
        Mexico_City("America/Mexico_City")
    }

    override fun get(province: String): String? {
        return timezones.get(province)
    }

    override fun getCapital(): String {
        return Timezones.Mexico_City.timezone
    }

    companion object {

        private val timezones = Tri<String>()

        init {
            for (timezone in Timezones.values()) {
                timezones.insert(timezone.name, timezone.timezone)
            }
        }
    }
}
